pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -36,42 +37,68 @@ const (
 )
 
 // MakeVirtualService creates a VirtualService from a Route object.
-func MakeVirtualService(route *v1alpha1.Route) (*networking.VirtualService, error) {
-	hostDomain := route.Spec.Domain
-	if route.Spec.Hostname != "" {
-		hostDomain = route.Spec.Hostname + "." + route.Spec.Domain
+func MakeVirtualService(routes []*v1alpha1.Route) (*networking.VirtualService, error) {
+	if len(routes) == 0 {
+		return nil, errors.New("routes must not be empty")
 	}
 
-	httpRoute, err := buildHTTPRoute(route)
+	namespace := routes[0].Namespace
+	hostname := routes[0].Spec.RouteSpecFields.Hostname
+	domain := routes[0].Spec.RouteSpecFields.Domain
+	urlPath := routes[0].Spec.RouteSpecFields.Path
+
+	var (
+		ownerRefs []metav1.OwnerReference
+		labels    = map[string]string{}
+		appNames  []string
+	)
+	for _, route := range routes {
+		// Each route will own the VirtualService. Therefore none of them can be a
+		// controller.
+		ownerRef := *kmeta.NewControllerRef(route)
+		ownerRef.Controller = nil
+		ownerRef.BlockOwnerDeletion = nil
+		ownerRefs = append(ownerRefs, ownerRef)
+
+		// Labels
+		labels = resources.UnionMaps(route.GetLabels(), labels)
```

Comment:
I don't think it makes sense to union the labels from all routes because they might contain ACL info and/or app-specific info. Maybe re-create a set just for the VirtualService with what we expect to be there?